Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3386,17 +3389,33 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
     if (!visibleRow)
         rectsInPluginCoordinates.reserveCapacity(findMatchRects.size());
 
+    auto unobscuredContentRectInPluginSpace = [this] -> std::optional<IntRect> {
+        if (!m_frame || !m_frame->page() || !m_frame->coreLocalFrame())
+            return { };
+        RefPtr view = m_frame->coreLocalFrame()->view();
+        if (!view)
+            return { };
+        return convertFromRootViewToPlugin(view->unobscuredContentRect());
+    }();
+
     for (auto& perPageInfo : findMatchRects) {
         if (visibleRow && !visibleRow->containsPage(perPageInfo.pageIndex))
             continue;
 
         auto pluginRect = convertUp(CoordinateSpace::PDFPage, CoordinateSpace::Plugin, perPageInfo.pageBounds, perPageInfo.pageIndex);
-        rectsInPluginCoordinates.append(pluginRect);
+        if (!unobscuredContentRectInPluginSpace || pluginRect.intersects(unobscuredContentRectInPluginSpace.value()))
+            rectsInPluginCoordinates.append(pluginRect);
     }
 
     return rectsInPluginCoordinates;
 }
 
+HashMap<WebFoundTextRange::PDFData, RetainPtr<PDFSelection>>& UnifiedPDFPlugin::webFoundTextRangePDFDataSelectionMap()
+{
+    static NeverDestroyed<HashMap<WebFoundTextRange::PDFData, RetainPtr<PDFSelection>>> map;
```

Comment:
Wouldn't it make more sense for this to be per-instance rather than static?

Reply:
I think we'd get similar performance characteristics either way (as long as we manage the cache properly), but you're correct that conceptually this map belongs to a single Unified PDF instance.

Reply:
Let me change this map to a data member.